### ISSUE TITLE
ClusterLoad: timeout is not enforced while determining the cluster IOPS limit

### ISSUE THREAD

**Issue by fbalak** [red-hat-storage/ocs-ci#4372]
Cluster load (https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/ocs/cluster_load.py) should run in separate thread. Now it can delay some tests for hours. E.g. `test_load_crush_map` test was delayed by 6 hours in https://ocs4-jenkins-csb-ocsqe.apps.ocp4.prod.psi.redhat.com/job/qe-deploy-ocs-cluster-prod/1003/consoleFull. In the run the `test_load_crush_map` test failed due to ceph health error that was caused by utilizing all osds by background io.
```
ocs_ci.ocs.exceptions.CephHealthException: Ceph cluster health is not OK. Health: HEALTH_ERR 3 full osd(s); 3 pool(s) full
```

**Comment by petr-balogh** [comment]
@ebenahar can you please add your input here?

**Comment by ebenahar** [comment]
Hi @fbalak @petr-balogh Cluster load is already done in a separate thread - https://github.com/red-hat-storage/ocs-ci/blob/352e0403eb28889a9a87aabdc3776501d5e8cfdd/tests/conftest.py#L1287

But that happens only after the preparation of cluster load is done. Before that, the preparation indeed takes place as part of the main thread.

The preparation of cluster load has a timeout of 30 minutes. I see that in this particular run, it took way longer than that. So I assume that there is a problem with the code. 

I will check

**Comment by petr-balogh** [comment]
I see another occurrence here:
https://ocs4-jenkins-csb-ocsqe.apps.ocp4.prod.psi.redhat.com/job/qe-deploy-ocs-cluster-prod/1011/testReport/tests.ecosystem.upgrade/test_configuration/

https://ocs4-jenkins-csb-ocsqe.apps.ocp4.prod.psi.redhat.com/job/qe-deploy-ocs-cluster-prod/1011/

The test_load_crush_map took 4:41 .

In 4.7 upgrade it took ~50 mins here:
https://ocs4-jenkins-csb-ocsqe.apps.ocp4.prod.psi.redhat.com/view/Upgrade-OCS/job/qe-trigger-aws-ipi-3az-rhcos-3m-3w-upgrade-ocs-auto/29/testReport/tests.ecosystem.upgrade/test_configuration/

and here:
https://ocs4-jenkins-csb-ocsqe.apps.ocp4.prod.psi.redhat.com/job/qe-trigger-aws-ipi-fips-encryption-3az-rhcos-3m-3w-3i-upgrade-ocs-auto/7/testReport/tests.ecosystem.upgrade/test_configuration/

Only 13 mins.

**Comment by petr-balogh** [comment]
@fbalak @ebenahar as it's affecting the upgrade a lot and even on AWS because of this the execution took more than 36 hours and cluster go deleted almost at the end of execution so the whole execution failed and didn't even send email from tier1 after upgrade.  @dahorak will open the issue for that, but we need to find the root cause of this ASAP and fix it.

**Comment by fbalak** [comment]
The issue is present in run 1630404008. The test `test_load_crush_map` runs for 7 hours and fails with ceph error:
```
ocs_ci.ocs.exceptions.CephHealthException: Ceph cluster health is not OK. Health: HEALTH_ERR 3 full osd(s); 3 pool(s) full
```
In the test is initiated `ClusterLoad`.

**Comment by petr-balogh** [comment]
@fbalak if you are referring to this run: 
https://ocs4-jenkins-csb-ocsqe.apps.ocp4.prod.psi.redhat.com/job/qe-deploy-ocs-cluster-prod/1731/

Than it was running from release-4.6 branch which doesn't have this fix cherry-picked.

**Comment by fbalak** [comment]
@petr-balogh Ok, closing for now
